### PR TITLE
♻️ Convert scripts and migrations to Typescript

### LINKED ISSUE #301: ✒️ script files written in typescript
We should take advantage of our type system for scripts. Script files should be written in typescript then ran from the build folder in js.

### PR DESCRIPTION
**Pull request by eliascodes**
Fixes #301 

### Changes
* ♻️ Convert all scripts, migrations and seeds to TS
* 🔧 Separate out build stages on Travis

### CODE REVIEW

**Comment by eliascodes** [comment]
Looks bigger than it really is.

File: database/migrations/20180901_15_run_initial_seeds.ts [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
 
-exports.up = async (knex) => {
+exports.up = async (knex: Knex) => {
   const NODE_ENV = getEnvironment(process.env.NODE_ENV);
   const basePath = path.resolve(__dirname, '..', 'seeds', NODE_ENV);
   const files = fs.readdirSync(basePath);
   const fns = files.sort().map((file) => require(path.join(basePath, file)).seed);
 
-  for (let i = 0; i < fns.length; i++) {
+  for (let i = 0; i < fns.length; i = i + 1) {
```
Comment (astroash) [inline database/migrations/20180901_15_run_initial_seeds.ts]: ?

Comment (eliascodes) [reply database/migrations/20180901_15_run_initial_seeds.ts]: 🤷🏼‍♂️ our tslint conf doesn't like `++`

**Comment by eliascodes** [comment]
Note: Before this is deployed to `master`, the `knex_migrations` table needs to be updated to prevent build failure:

```sql
UPDATE knex_migrations SET name = REPLACE(name, '.js', '.ts');
```

**Pull request merged**